ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/db/DefaultMutableBlockchain.java

```diff
@@ -51,10 +53,23 @@
   private final Subscribers<BlockAddedObserver> blockAddedObservers = new Subscribers<>();
 
   public DefaultMutableBlockchain(
-      final Block genesisBlock, final BlockchainStorage blockchainStorage) {
+      final Block genesisBlock,
+      final BlockchainStorage blockchainStorage,
+      final MetricsSystem metricsSystem) {
     checkNotNull(genesisBlock);
     this.blockchainStorage = blockchainStorage;
     this.setGenesis(genesisBlock);
+
+    metricsSystem.createGauge(
+        MetricCategory.BLOCKCHAIN,
+        "height",
+        "Height of the chainhead",
+        () -> (double) this.getChainHeadBlockNumber());
+    metricsSystem.createGauge(
+        MetricCategory.BLOCKCHAIN,
+        "difficulty_total",
+        "Total difficulty of the chainhead",
+        () -> (double) this.getChainHead().getTotalDifficulty().toLong());
```

Comment:
This cast to double is proving to be an annoyingly common pattern.  Should we introduce a `LongSupplier` version of gauges?  Even if it just did the cast internally it would tidy code up.  Also, in hind-sight I should have used `DoubleSupplier`.  Might be a separate PR...

Reply:
Yes, because I also see the potential of a UInt256Supplier.